projects/client/RabbitMQ.Client/src/client/api/IModel.cs

```diff
@@ -226,6 +226,11 @@ public interface IModel : IDisposable
         void BasicPublish(string exchange, string routingKey, bool mandatory,
             IBasicProperties basicProperties, byte[] body);
 
+        [AmqpMethodDoNotImplement(null)]
+        void BasicBatchPublish(string exchange, string routingKey, bool mandatory,
```

Comment:
IEnumerable is great multi purpose when you look at it from a caller perspective. One of the downsides it has it requires you to be aware of multiple enumerations, you cannot access things like count or length without using LINQ (allocs) and it requires the lib to copy around things into internal materialized data structures. 

If the API is considered lower level I would lean towards using an array if possible but like I said IEnumerable offers more convenience from the caller perspective. Just some food for thought

Reply:
It's generally OK if the implementation ensure that it only enumerates once. To do a count, it can enumerate into, e.g. a `List<T>`, and inspect the `Count` property.

Reply:
Sorry, I see what you're saying, even a `ToList()` results in allocations internally. This is a valid concern for a hot code path.